core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
Could you remind me of what the problem is here?  If we come here for a Level 1 instr where we only know the length and we have not decoded it beyond that, explaining the OP_UNDECODED, the up-decode will fail if our decoder thinks it's invalid, but why is that a problem?  Won't we continue and encode its still-valid raw bits?